### PR TITLE
parser: tidy up parameter/message parsing

### PR DESCRIPTION
**Pull request by pdevine**
This change addresses how parameters and messages are handled while parsing a Modelfile. Currently a `MESSAGE` command creates a string that looks like "<role>: <message>" which then has to be re-parsed, and `PARAMETER` ends up being the default data type which makes it difficult to add other multi-part commands.

This change introduces a Message and a Parameter type which properly handle properties such as the role and the name of the parameter.

### CODE REVIEW

File: parser/parser.go [hunk 1]
```diff
@@ -62,14 +62,15 @@ func (f Modelfile) CreateRequest(relativeDir string) (*api.CreateRequest, error)
 	for _, c := range f.Commands {
 		switch c.Name {
 		case "model":
-			path, err := expandPath(c.Args, relativeDir)
+			name := c.Args.(string)
```
Comment (jmorganca) [inline parser/parser.go:65]: Should we type check this to avoid a panic?

File: parser/parser.go [hunk 2]
```diff
@@ -321,12 +339,16 @@ func (c Command) String() string {
 	case "model":
 		fmt.Fprintf(&sb, "FROM %s", c.Args)
 	case "license", "template", "system", "adapter":
-		fmt.Fprintf(&sb, "%s %s", strings.ToUpper(c.Name), quote(c.Args))
+		data := c.Args.(string)
```
Comment (jmorganca) [inline parser/parser.go:342]: Similar to above, it seems likely this might panic if the input type is wrong

**Review by jmorganca — COMMENTED** [review]
This might need some more type checks. I'm also worried the tests may not fully cover edge cases (e.g. `"` quotes in a PARAMETER, etc) but I didn't look into detail
